internal/services/automation/resourceids.go

```diff
@@ -0,0 +1,3 @@
+package automation
+
+//go:generate go run ../../tools/generator-resource-id/main.go -path=./ -name=AutomationJobSchedule -id=/subscriptions/12345678-1234-9876-4563-123456789012/resourceGroups/group1/providers/Microsoft.Automation/automationAccounts/account1/runBook/book1/schedule/schedule1
```

Comment:
The functionality provided by `resourceids.go` shouldn't be used for new resources and will be removed once we've finished migrating off them - instead, since this is already using `hashicorp/go-azure-sdk` - the relevant Resource ID from `hashicorp/go-azure-sdk` should be used instead

Reply:
the Resource ID in `go-azure-sdk` cannot work as expect in the issue of this PR is tring to fix. because the job schedule id will be updated automatically when the runbook resource updated. so I create this new job schedule ID Format to represent the link between the runbook and schedule.

Reply:
I updated this PR to fix some logic and tested it locally. Although the ID format of the resource has changed, I verified that this update should not introduce any breaking changes.